src/cgame/cg_weapons.cpp

```diff
@@ -1328,10 +1325,9 @@ static void CG_CalculateWeaponPosition( vec3_t out_origin, vec3_t out_angles )
 	const float scaleY = 2e-3;
 
 	VectorAdd( angles, offsets.bob, angles );
-	origin += up * atanf( offsets.angvel[ 0 ] * scaleY ) * limitY;
-	origin += right * atanf( offsets.angvel[ 1 ] * scaleX ) * limitX;
+	VectorMA( cg.refdef.vieworg, atanf( offsets.angvel[ 0 ] * scaleY ) * limitY, cg.refdef.viewaxis[ 2 ], out_origin );
+	VectorMA( out_origin, atanf( offsets.angvel[ 1 ] * scaleX ) * limitX, cg.refdef.viewaxis[ 1 ], out_origin );
```

Comment:
maybe it would be a good idea to use some `vec3_t& up, right;` here to avoid obfuscating the code more than it already is. (I don't think it's required either though)

Reply:
All this will soon be rewritten to use glm anyway. This PR is "just" to make the glm migration one easier to do (because of the power of sed) and review.